src/Features/Core/Portable/StackTraceExplorer/ParsedStackFrame.cs

```diff
@@ -92,157 +75,184 @@ internal sealed class ParsedStackFrame : ParsedFrame
                 var members = type.GetMembers();
```

Comment:
This comment refers to code above this in the collapsed region.

1. you cannot call GetRquiredCompilationAsync. It will crash on non VB/C# projects.  You need to check project.SupportsCompilation.
2. getting a compilation is super expensive (requires parsing all files).  Prior to doing this, we should see if you can first see if the project contains the type name in question (this can be done using some of hte indices available to us).

Reply:
> Prior to doing this, we should see if you can first see if the project contains the type name in question

Would this just be a `project.ContainsSymbolAsync` call?

Reply:
taking offline.

Reply:
This now uses `project.ContainsSymbolsWithNameAsync` to match type names.

Reply:
note: tht only works for source symbols.  so any stack frames in metadata types won't work (which seems bad).

Reply:
Does `GetTypeByMetadataName` still get metadata types? Is there a good way to check for metadata types without a compilation?

Reply:
yes... yes they do. 

I added a multipass approach where the "fast" pass uses `ContainsSymbolsWithNameAsync` before getting a compilation, and anything not checked will still have to get a compilation to check metadata symbols (I think?).

Also added a test.

Reply:
Note. Symboltreeinfo can be used to index metadata.